### PR TITLE
EdgeX Liaison Section

### PR DESCRIPTION
**Pull request by mmccool**
Description of EdgeX and WoT's relationship to it.  Note that this should be carefully reviewed for accuracy, we don't want to misrepresent a potential partner's project.  In particular, we need to check if they have adopted another mechanism for formally expressing metadata (e.g. OpenAPI) since I last looked at it in detail.

Also, EdgeX is basically an example of a software stack for a "hub" architecture.  There are several other examples of such architectures, including WebThings, SmartThings, Hubicity, etc.  So I have listed some related issues in architecture (see below) that relate to this.  Also, it seems to me that if we are going to talk about EdgeX we probably want to also talk about WebThings and maybe mention some of the other ones (there is a list in the issues of the wot-security repo).

Related issues:
- https://github.com/w3c/wot-usecases/issues/45
- https://github.com/w3c/wot-architecture/issues/585
- https://github.com/w3c/wot-architecture/issues/581

***
<a href="https://pr-preview.s3.amazonaws.com/mmccool/wot-usecases/pull/158.html" title="Last updated on Dec 14, 2021, 1:16 PM UTC (3d7f5ad)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/wot-usecases/158/4a01cdc...mmccool:3d7f5ad.html" title="Last updated on Dec 14, 2021, 1:16 PM UTC (3d7f5ad)">Diff</a>

### CODE REVIEW

**Comment by mmccool** [comment]
For reviews, Farshid would make sense (since he now works for EdgeX) but github isn't letting me add him since it seems he have never commented on this repo.  I will email him directly.

File: index.html [hunk 1]
```diff
@@ -7043,11 +7046,23 @@ <h2>ISO TC184/SC4 (Sebastian/Kaz)</h2>
...
-        Scope of the collaboration.
-        Provide Website link.
+      <h2>EdgeX Foundry</h2>
+      <p>
+        The EdgeX Foundry [EDGEX] is a community-driven open source project,
+        organized under the Linux Foundation, to build
+        a secure software stack for IoT hubs. Its goal is to enable interoperability
+        by combining a set of key IoT services with a set of interfaces to a variety
+        of IoT device protocols and ecosystems.  This organization focuses on software
+        development rather than standards development per se.
```
Comment (farshidtz) [inline index.html:7056]: I'd say the EdgeX goal is to create a reference architecture. The developed software is a reference implementation of that architecture.

**Review by farshidtz — COMMENTED** [review]
@mmccool I do work on (not for) EdgeX but do not represent the organization. With that said, I think your text about not following a metadata standard for describing device interfaces is accurate.

The metadata schema can be explored from:
- OpenAPI spec: https://github.com/edgexfoundry/edgex-go/blob/1876cd192641e06e6c48213757fa64687e1ce8f0/openapi/v2/core-metadata.yaml#L118
- Documentation: https://docs.edgexfoundry.org/2.1/microservices/core/metadata/Ch-Metadata/

File: index.html (hunk 1 above)
Comment (mmccool) [reply index.html:7056]: I'm going to steal your text and change the last sentence to: "The EdgeX goal is to create a reference architecture, and there is a reference implementation of that architecture."

Comment (mmccool) [reply index.html]: made a few more changes than just changing the last sentence to make things consistent, but I think I've captured your point and am going to mark this resolved.

**Comment by mmccool** [comment]
Also, regarding metadata, last I looked EdgeX did not necessarily expose direct interfaces to devices, e.g. via a standardized proxy, but instead had (relatively fixed) metadata for an overall generic API.  So I will have to look again.  What I was thinking though is that the current overall API makes sense, but we add a WoT Directory service to that, and THAT service is a standard WoT service for Things (that exposes TDs, since it would be an instance of a WoT TDD).

**Comment by mmccool** [comment]
Also, I find it interesting that in the SmartThings system I am playing with, the "hub" also shows up as a "device" on my dashboard.  In WoT terms, this means the hub is also a Thing.  So maybe the overall generic interface to the hub could also be expressed as a WoT TD.  I will have to look at the OpenAPI definition.  Conversely, expressing the WoT TDD API in OpenAPI would be an interesting exercise.

**Comment by mmccool** [comment]
Anyway, the PR is ready to merge, I think.  The above thoughts are more related to how we move forward with EdgeX (and possibly also talk about hubs in Architecture).

**Comment by mmccool** [comment]
@farshidtz did you want to be listed as a contributor to this use case?  If so, let me know.  I think we should merge this now, but we can add you later.

**Comment by farshidtz** [comment]
Thanks for asking @mmccool. It's not needed as I only provided minor feedback. I'll consider contributing more in future.

**Pull request merged**